### DISCUSSION TITLE
Can I use rabies on an M1 Mac?

### DISCUSSION THREAD

**Discussion by colon-perez** [CoBrALab/RABIES#298] [General]
I am trying to install RABIES but as I start downloading dependencies etc. I keep hitting a roadblock with some tools because of ARM chips. I was going to go with singularity, but that one won't work, but then it dawned on me that it may not work at all. Am I missing something? Maybe help to get this started

**Comment by gdevenyi** [comment]
Use singularity or docker. Both should be runnable in the transparent rosetta emulator.

**Reply by colon-perez** [reply]
Thanks, is there a tutorial (new to singularity and containers in general). 

When I tried singularity like this
singularity build rabies.sif docker://gabdesgreg/rabies:tagname

I get this
WARNING: Authentication token file not found : Only pulls of public images will succeed
FATAL:   Only remote builds are supported on this platform

It seems this is an M1 issue but was wondering if someone has find work around or can you point me to the rosetta emulator?

**Reply by Gab-D-G** [reply]
This is because 'tagname' does not refer to an actual tagname. Select an appropriate tag from the current versions available on docker hub (https://hub.docker.com/r/gabdesgreg/rabies/tags). I would recommend 0.4.8 or 0.5.0-dev.

**Reply by colon-perez** [reply]
Thanks, but I still get the same error when I use either tag. I went with Singularity because it seems a more robust way but I may try to install docker and see if it works out. I did use pip install rabies and that one works but I guess I am missing dependencies because I get errors when I try to use it. Is there a place where I can see what I need to get it working without the container? I am trying to use the test data set from the website, so the issues are with install or related.

**Reply by Gab-D-G** [reply]
Hi, all I could find related to your issue with Singularity is this https://github.com/vatlab/sos/issues/1352#issuecomment-569588430, it seems like `singularity build` may not yet work on Mac platform... Hopefully Docker works. Otherwise, the list of dependencies for a local installation canbe found here https://github.com/CoBrALab/RABIES/blob/master/dependencies.txt. However, some of those (if not all) will require linux platform I believe, and will likely be more trouble than figuring out Docker.

**Reply by colon-perez** [reply]
I came to the same conclusion and have been trying to get docker going. But I get this.

Unable to find image 'rabies:latest' locally
docker: Error response from daemon: pull access denied for rabies, repository does not exist or may require 'docker login': denied: requested access to the resource is denied.

I am using the command 
docker run -it --rm \
-v $PWD/test_dataset:/input_BIDS:ro \
-v $PWD/preprocess_outputs:/preprocess_outputs/ \
rabies -p MultiProc preprocess /input_BIDS/ /preprocess_outputs/ --apply_STC --TR 1.2 masking=true,brain_extraction=false,template_registration=SyN,fast_commonspace=false

Should there be another build command after pull? I can see the rabies image (gabdesgreg/rabies) on my docker desktop  but I get the error above.

**Reply by gdevenyi** [reply]
Another instance of :latest not being a tag in our docker. Will be fixed soon by things I'm working on.

The latest available tag is 0.5.0-dev, you should probably use 0.4.8

**Reply by colon-perez** [reply]
Thanks for all your help! Now it's finally running. A novice on Docker and Apple M1 processor, and you have a mix for disaster. I thought that once you pull the image of rabies then docker would use the one on the desktop but I state the entire option and now it works. Maybe it can be noted that you need to activate rosetta on docker and stipulate the platform as amd. Also the command for docker on the "running the software" was missing the "commonspace_reg", options were there but not the flag. I highlighted in BOLD what I needed to do in case someone encounters the same problem.
 
docker run -it --rm **--platform linux/amd64** \
-v $PWD/test_dataset:/input_BIDS:ro \
-v $PWD/preprocess_outputs:/preprocess_outputs/ \
**gabdesgreg/rabies:0.4.8** -p MultiProc preprocess /input_BIDS/ /preprocess_outputs/ --apply_STC --TR 1.2 **--commonspace_reg** masking=true,brain_extraction=false,template_registration=SyN,fast_commonspace=false
